Mathlib/RingTheory/Finiteness/Quotient.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2020 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen, Yongle Hu
+-/
+import Mathlib.Algebra.Group.Subgroup.Actions
+import Mathlib.RingTheory.FiniteType
+import Mathlib.RingTheory.Ideal.Pointwise
+import Mathlib.RingTheory.Ideal.Over
+
+/-!
+# Finiteness of quotient modules
+-/
+
+variable (R : Type*) [CommSemiring R] {A B C : Type*} [CommRing A] [CommRing B] [CommRing C]
+  [Algebra A B] [Algebra A C] [Algebra R A] [Algebra R B] [IsScalarTower R A B]
+  (P : Ideal B) {Q : Ideal C} (p : Ideal A) [Q.LiesOver p] [P.LiesOver p]
+  (G : Type*) [Group G] [MulSemiringAction G B] [SMulCommClass G A B]
```

Comment:
Looks like some of these aren't needed: `G`, for example? It also appears that R might not be needed either, but I'm not certain.